Sources/Disposable.swift

```diff
@@ -202,41 +160,43 @@ public final class CompositeDisposable: Disposable {
 		}
 	}
 
-	/// Add the given disposable to the list, then return a handle which can
-	/// be used to opaquely remove the disposable later (if desired).
+	/// Add the given disposable to the composite.
 	///
 	/// - parameters:
-	///   - d: Optional disposable.
+	///   - disposable: A disposable.
 	///
-	/// - returns: An instance of `DisposableHandle` that can be used to
-	///            opaquely remove the disposable later (if desired).
+	/// - returns: A disposable to remove `disposable` from the composite. `nil` if the
+	///            composite has been disposed of, `disposable` has been disposed of, or
+	///            `disposable` is `nil`.
 	@discardableResult
-	public func add(_ d: Disposable?) -> DisposableHandle {
-		guard let d = d else {
-			return DisposableHandle.empty
+	public func add(_ disposable: Disposable?) -> Disposable? {
```

Comment:
We could leave this as non-optional by returning a disposable that does nothing. I'm not sure which would be better. Is there a particular reason you decided to return an optional?